include/libdnf/utils/utils.hpp

```diff
@@ -70,6 +72,20 @@ enum class GoalProblem : uint32_t {
     NOT_FOUND_IN_REPOSITORIES = (1 << 5)
 };
 
+enum class GoalSetting { AUTO, SET_TRUE, SET_FALSE };
+
+struct GoalSettings {
```

Comment:
This code shouldn't be in a header called `include/libdnf/utils/utils.hpp`, in fact I'd argue we shouldn't have such header file at all.

The code here looks like it should much rather be in a file called e.g. ` include/libdnf/base/goal_settings.hpp`